### PR TITLE
Non mandatory ZipCode in US checkout

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

We would like to make the US ZipCode field non-mandatory (ie Displayed but not required to be filled in to checkout).  Currently this affects->
1. US Contributions : Contribution Checkout
2. US Recurring S+ Contributions : Contribution Checkout
3. US Recurring Contribution: Generic Checkout

This is the client-side aspect ONLY. Server-side aspects to be determined in follow-up PR.

## Why are we doing in this?

As part of a change we made in Sep’23 we changes both the position of the Zipcode and made it mandatory , that has lead to an issue with Apple and Google pay payment in the US , as in these fail when the mandory field is not filled.

[**Trello Card**](https://trello.com/c/Cpvg3n5l/896-zipcode-make-it-non-mandatory-on-the-checkout-page-for-single-recurring-and-s-for-us)

### CODE REVIEW

**Review by jamesgorrie — APPROVED** [review]

File: support-frontend/assets/pages/supporter-plus-landing/setup/legacyActionCreators.ts [hunk 1]
```diff
@@ -221,7 +220,7 @@ function regularPaymentRequestFromAuthorisation(
 				? billingState
 				: null,
 			// required Zuora field if country is US or CA
-			postCode: billingCountry === 'US' ? postCode : null,
+			postCode: null,
```
Comment (paulbrown1982) [inline support-frontend/assets/pages/supporter-plus-landing/setup/legacyActionCreators.ts:223]: Can we still provide it to Zuora/Stripe if it is given?

Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/setup/legacyActionCreators.ts:223]: Good point, will test that now on Code and re-enable. Thanks.

Comment (paulbrown1982) [reply support-frontend/assets/pages/supporter-plus-landing/setup/legacyActionCreators.ts:223]: Re: https://stripe.com/en-no/guides/optimizing-authorization-rates#:~:text=In%20particular%2C%20passing%20the%20ZIP,the%20customer%20and%20how%20much.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/setup/legacyActionCreators.ts]: Yes, tested  and this is back in place.

**Pull request merged**